src/components/screens/monitor/delegates/withForgingStatus.js

```diff
@@ -72,35 +72,46 @@ const withForgingStatus = delegatesKey => (ChildComponent) => {
       );
       const height = blocks[0] && blocks[0].height;
       this.setState({
-        nextForgers: nextForgers.reduce((accumulator, delegate, i) => ({
-          ...accumulator,
-          [delegate.publicKey]: {
-            forgingTime: moment().add(i * 10, 'seconds'),
-            nextHeight: height + i + 1,
-          },
-        }), {}),
+        nextForgers: nextForgers.map((delegate, i) => ({
+          ...delegate,
+          forgingTime: moment().add(i * 10, 'seconds'),
+          nextHeight: height + i + 1,
+          username: delegate.username,
+        })),
       });
     }
 
     componentDidUpdate(prevProps) {
       const { latestBlocks } = this.props;
       const newBlock = latestBlocks[0] || {};
+      const forgerIndex = this.state.nextForgers.findIndex(delegate =>
```

Comment:
💡 IMO the code (here and in `getUpdatedNextForgersList`) would be more straight-forward with
```suggestion
      const forger = this.state.nextForgers.find(delegate =>
```